modules/core/src/main/scala/iota/Cop.scala

```diff
@@ -68,6 +68,12 @@ object Cop {
       else None
     def apply(a: A): Cop[L] = inj(a)
     def unapply(c: Cop[L]): Option[A] = proj(c)
+
+    def fold(c: Cop[L]): Either[Cop[TList.Op.Remove[A, L]], A] =
```

Comment:
We could potentially also create `RemoveL`: 

```scala
final class RemoveL[F[_], L <: KList] private[RemoveL](index: Int) {
  def apply[A](c: CopK[L, A]): Either[CopK[KList.Op.Remove[F, L], A], F[A]] =
    Either.cond(
      c.index == index,
      c.value.asInstanceOf[F[A]],
      new CopK(if (c.index < index) c.index else c.index - 1, c.value))    
}

object RemoveL {
  def apply[F[_], L <: KList](implicit ev: RemoveL[F, L]): RemoveL[F, L] = ev
  implicit def makeRemoveL[F[_], L <: KList](implicit ev: KList.Pos[L, F]): RemoveL[F, L] =
    new RemoveL[F, L](ev.index)
}
```

If we add more type operations to our coproducts, these probably need value level operations as well.

Too bad we would have to define these for `TList`, `KList` (`QList` ...).